nb-virtdata/virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_collection/Map.java

```diff
@@ -61,12 +62,23 @@ public Map(LongToIntFunction sizeFunc,
             "create a map of size 2, with a specific function for each key and each value"})
     @SafeVarargs
     public Map(LongFunction<Object>... objfuncs) {
-        this.mode = Mode.Tuples;
         if ((objfuncs.length%2)!=0) {
-            throw new RuntimeException("An even number of functions must be provided.");
+            Object testValue = objfuncs[0].apply(0L);
+            if (testValue instanceof Number n) {
+                LongFunction<Object>[] finalObjfuncs = objfuncs;
+                this.sizeFunc= l -> ((Number) finalObjfuncs[0].apply(l)).intValue();
+                objfuncs = Arrays.copyOfRange(objfuncs, 1, objfuncs.length);
+                this.mode=Mode.VarSized;
+            } else {
+                throw new RuntimeException("An even number of functions must be provided, unless "
+                                           + "the first one produces a numeric value.");
```

Comment:
Since we plan on making Runtime errors suggest GH issue submission this should throw a Config or Basic Error